api/rest/restapi_test.go

```diff
@@ -1073,21 +1074,27 @@ type httpTestcase struct {
 }
 
 func httpStatusCodeChecker(resp *http.Response, expectedStatus int) error {
-	if resp.StatusCode != expectedStatus {
-		return fmt.Errorf("bad HTTP status code: %d", resp.StatusCode)
+	if resp.StatusCode == expectedStatus {
+		return nil
 	}
-	return nil
+	return fmt.Errorf("unexpected HTTP status code: %d", resp.StatusCode)
 }
 
 func assertHTTPStatusIsUnauthoriazed(resp *http.Response) error {
 	return httpStatusCodeChecker(resp, http.StatusUnauthorized)
 }
 
-func assertHTTPStatusIsNotUnauthoriazed(resp *http.Response) error {
-	if assertHTTPStatusIsUnauthoriazed(resp) == nil {
-		return fmt.Errorf("unexpected HTTP status code: %d", http.StatusUnauthorized)
+func assertHTTPStatusIsTooLarge(resp *http.Response) error {
+	return httpStatusCodeChecker(resp, http.StatusRequestHeaderFieldsTooLarge)
+}
+
+func makeInvertedHTTPStatusAssert(checker responseChecker) responseChecker {
```

Comment:
```suggestion
func makeHTTPStatusNegatedAssert(checker responseChecker) responseChecker {
```
Just to be clear it is the boolean aspect of the assert that is being inverted and not the status code.